brayns/io/ColorMapLoader.cpp

```diff
@@ -0,0 +1,121 @@
+/* Copyright (c) 2015-2016, EPFL/Blue Brain Project
+ * All rights reserved. Do not distribute without permission.
+ * Responsible Author: Cyrille Favreau <cyrille.favreau@epfl.ch>
+ *
+ * This file is part of Brayns <https://github.com/BlueBrain/Brayns>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <brayns/common/log.h>
+#include <brayns/common/material/Texture2D.h>
+#include <brayns/common/scene/Scene.h>
+
+#include <fstream>
+
+#include "ColorMapLoader.h"
+
+namespace brayns
+{
+
+const uint DEFAULT_ALPHA = 255;
+
+ColorMapLoader::ColorMapLoader()
+{
+}
+
+const Vector4uis& ColorMapLoader::getValues() const
+{
+    return _values;
+}
+
+void ColorMapLoader::setValues( const Vector4uis& values )
+{
+    _values.clear();
+    for( auto value: values )
+        _values.push_back( value );
+}
+
+bool ColorMapLoader::loadFromFile( const std::string& filename )
+{
+    BRAYNS_INFO << "Loading color map from " << filename << std::endl;
+    std::ifstream file( filename, std::ios::in );
+    if( !file.good( ))
+    {
+        BRAYNS_ERROR << "Could not open file " << filename << std::endl;
+        return false;
+    }
+
+    bool validParsing = true;
+    std::string line;
+
+    while( validParsing && std::getline( file, line ))
+    {
+        std::vector< uint > lineData;
+        std::stringstream lineStream(line);
+
+        size_t value;
+        while( lineStream >> value )
+            lineData.push_back(value);
+
+        switch( lineData.size() )
+        {
+        case 3:
+            _values.push_back( Vector4ui(
+                lineData[0], lineData[1], lineData[2], DEFAULT_ALPHA ));
+            break;
+        case 4:
+            _values.push_back( Vector4ui(
+                lineData[0], lineData[1], lineData[2], lineData[3] ));
+            break;
+        default:
+            BRAYNS_ERROR << "Invalid line: " << line << std::endl;
+            validParsing = false;
+            break;
+        }
+    }
+
+    file.close();
+    return validParsing;
+}
```

Comment:
2 things:
- That's two new mandatory dependencies that I'd need to add in order to replace 20 lines of basic c++ code, you'll need to convince me.
- The existing implementation of the LookupTable1D is very 'livre' specific and does not currently satisfy my needs: typically additional parameters such as light emission and ambient occlusion strengths, and of course more than 256 values to match those of the simulation. It's already implemented in my branch, but I am pushing things little by little to make the reviewing easier. So let's first decide on the event, and then I will use it.